test/python/test_slice.py

```diff
@@ -150,3 +151,45 @@ def run_test(sess):
         # expected.append(inp[:, tf.newaxis])
         for v, e in zip(slice_vals, expected):
             np.testing.assert_array_equal(v, e)
+
+    def test_strided_slice_zerodim(self):
+        inp = np.random.rand(4, 0, 5).astype("f")
+        slice_ts = []
+        expected = []
+
+        a = np.array([float(x) for x in inp.ravel(order="C")])
+        a.shape = (4, 0, 5)
+
+        x = tf.placeholder(dtype=dtypes.float32)
+
+        #(slicing an empty dim by empty slice)
+        slice_ts.append(x[1:2, 2:2, 1:2])
+        #(slicing an empty dim by non empty slice)
+        slice_ts.append(x[1:2, 1:2, 1:2])
+
+        def run_test(sess):
+            return sess.run(slice_ts, feed_dict={x: a})
+
+        slice_vals = self.with_ngraph(run_test)
+
+        expected.append(inp[1:2, 2:2, 1:2])
+        expected.append(inp[1:2, 1:2, 1:2])
+
+        for v, e in zip(slice_vals, expected):
+            np.testing.assert_array_equal(v, e)
+
+    def test_incorrect_strided_slice(self):
+        inp = 0
+        slice_ts = []
+
+        x = tf.placeholder(dtype=dtypes.float32)
+
+        #(slicing an empty dim by empty slice)
+        slice_ts.append(x[1:1])
+
+        def run_test(sess):
+            return sess.run(slice_ts, feed_dict={x: inp})
+
+        with pytest.raises(Exception) as excinfo:
+            slice_vals = self.with_ngraph(run_test)
+            assert str(excinfo.value) == str(excinfo)
```

Comment:
I was trying something similar and found that assert inside with pytest.raises doesnt work. Would want to verify with you.
Right now the assert condition doesnt really check anything
We need to change it to:
```
assert "<error print>" in excinfo.value.message
```

Reply:
Changed this, please review.